process/sync/baseSync.go

```diff
@@ -197,6 +196,49 @@ func (boot *baseBootstrap) applyBlock(
 	return nil
 }
 
+func (boot *baseBootstrap) addHeadersToForkDetector(
+	shardId uint32,
+	nonce uint64,
+	getHeader func(uint32, uint64) (data.HeaderHandler, []byte, error)) {
+
+	header, headerHash, errNotCritical := getHeader(shardId, nonce)
+	if errNotCritical != nil {
+		log.Info(errNotCritical.Error())
+	}
+
+	prevHeader, prevHeaderHash, errNotCritical := getHeader(shardId, nonce-1)
```

Comment:
might add a test here not to compute 0-1 on uint64

Reply:
Done